appinfok8s.go

```diff
@@ -0,0 +1,60 @@
+package appwrap
+
+import (
+	"context"
+	"fmt"
+	"os"
+)
+
+type AppengineInfoK8s struct {
+	c context.Context
+}
+
+func (ai AppengineInfoK8s) AppID() string {
+	return os.Getenv("GOOGLE_CLOUD_PROJECT")
+}
+
+func (ai AppengineInfoK8s) InstanceID() string {
+	return os.Getenv("K8S_POD")
+}
+
+// ModuleHostname in K8s doesn't yet handle versions
+func (ai AppengineInfoK8s) ModuleHostname(version, module, app string) (string, error) {
+	if module == "" {
+		module = ai.ModuleName()
+	}
+	if app == "" {
+		app = ai.AppID()
+	}
+
+	domain := os.Getenv("K8S_DOMAIN")
+
+	return fmt.Sprintf("%s-dot-%s.%s", module, app, domain), nil
+}
+
+func (ai AppengineInfoK8s) ModuleName() string {
+	return os.Getenv("K8S_SERVICE")
+}
+
+func (ai AppengineInfoK8s) VersionID() string {
+	return os.Getenv("K8S_VERSION")
+}
+
+func (ai AppengineInfoK8s) Zone() string {
+	return os.Getenv("K8S_ZONE")
+}
+
+// ModuleHasTraffic not implemented in K8s
+func (ai AppengineInfoK8s) ModuleHasTraffic(moduleName, moduleVersion string) (bool, error) {
```

Comment:
We will have to see what this means - if it means "could receive traffic" as in the service readiness probe is healthy - then that is easy info to get. If it means we are actively receiving traffic, that is more difficult; but possible

Reply:
We're using this for the latter, so this would depend on how we do traffic splitting

Reply:
Yeah, if we decide to use istio, might have to query the virtual service here for traffic percentage to this version.